### PR TITLE
CONSOLE-3766: Update i18next-parser dev dependency in console

### PR DESCRIPTION
**Pull request by cyril-ui-developer**

### CODE REVIEW

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by jhadvig** [comment]
QE Approver:
/assign @yanpzhan 
Docs Approver: 
/assign @opayne1 
PX Approver:
/assign @RickJWagner

File: dynamic-demo-plugin/i18next-parser.config.js [hunk 1]
```diff
@@ -4,5 +4,9 @@ module.exports = {
   locales: ['en'],
   namespaceSeparator: '~',
   reactNamespace: false,
-  useKeysAsDefaultValue: true,
+  defaultValue: function (_, _, key, _) {
+    // The `useKeysAsDefaultValues` option is deprecated in favor of `defaultValue` option function arguments.
+    // The `key` is used to set default value.
+    return key;
+  }
```
Comment (TheRealJon) [inline dynamic-demo-plugin/i18next-parser.config.js:11]: Why is this needed here but not in the core app?

**Review by TheRealJon — COMMENTED** [review]
Just one question

File: dynamic-demo-plugin/i18next-parser.config.js (hunk 1 above)
Comment (cyril-ui-developer) [reply dynamic-demo-plugin/i18next-parser.config.js:11]: > Why is this needed here but not in the core app?

@TheRealJon The core app make use of the i18next-parser with a custom [script](https://github.com/openshift/console/blob/master/frontend/i18n-scripts/set-english-defaults.js) for parsing the strings. If we update the core app `defaultValue` with the function arguments, it would we'll have return empty string - `''` not `key` since the custom script handle that and more parser functions. So it is cleaner to assign `''` to `defaultValue` in core app.

**Comment by yanpzhan** [comment]
@XiyunZhao Pls help to test the pr, thanks!

**Comment by XiyunZhao** [comment]
This PR has been tested in manual and automation testing, no issue was found/
/label qe-approved
